tests/golang-regex-validator/test_regex_compiler.py

```diff
@@ -0,0 +1,32 @@
+import subprocess
+from os import path
+from pathlib import Path
+
+import pytest
+
+
+_FOLDER = Path(path.abspath(__file__)).parent
+_GO_TOOL_FILE = _FOLDER.parent / "regexCompiler.go"
+
+INVALID_REGEX = "[\\]"
+
+
+def invoke(input: str) -> subprocess.CompletedProcess:
+    return subprocess.run(["go", "run", _GO_TOOL_FILE], input=input, capture_output=True, encoding="utf-8")
+
+
+@pytest.mark.parametrize(
+    "regex,valid",
+    [
+        ("ab", True),
```

Comment:
Running `pytest` locally fails here with `no such file or directory`. The risks of pipeline not running the tests? :)

Reply:
True, the error is bigger, as we are pointing the the wrong file and some tests still pass :)

Reply:
Added test for that. Hope it is better now

Reply:
I have created https://github.com/RedHatInsights/insights-operator-gathering-conditions/pull/30 to add CI test job. We can merge that one first and the run tests for this one.

Reply:
Yes, but right now there is a test failing that indicates clearly what the problem is. 

> If you really wanted to fix that you could replace the weak assertion of assert cp.stderr != None with one that would check a specific part of the message (e.g. the regex string).

True, probably worth doing. However, note that the test with `assert cp.returncode == 0 if valid is True else cp.returncode != 0` in the invalid regex case is passing but not because the regex is invalid.

Anyways that will go away soon. I believe the way to expose the regex tool via a fixture. That way, the fixture should fail if the file does not exist and all dependent test won't run and their status will be a failure.

Reply:
> I believe the way to expose the regex tool via a fixture.

Yep, I've also converged to passing fixed repo paths to tests via fixtures in my tests.

Reply:
> If you really wanted to fix that you could replace the weak assertion of assert cp.stderr != None with one that would check a specific part of the message (e.g. the regex string).

I do not want to validate the regex, as it might be that quotation or character escaping changes something, but I have done some middle ground proposal https://github.com/RedHatInsights/insights-operator-gathering-conditions/pull/34 to check a message that we expect there. Hope this is an improvement.